rclcpp_lifecycle/test/test_state_wrapper.cpp

```diff
@@ -37,15 +37,15 @@ TEST_F(TestStateWrapper, wrapper) {
   }
 
   {
-    rcl_lifecycle_state_t lc_state = {"my_c_state", 2, NULL, NULL, 0};
+    rcl_lifecycle_state_t lc_state = {const_cast<char *>("my_c_state"), 2, NULL, NULL, 0};
```

Comment:
Why is this `const_cast` necessary?

Same below.

Reply:
C++11 raises a warning here, that a const char literal is assigned to a non-const.

https://stackoverflow.com/questions/20944784/why-is-conversion-from-string-constant-to-char-valid-in-c-but-invalid-in-c
Having it cast to a non-const seems to be a valid solution. Not sure if it's the most elegant one though.

Reply:
I think the question is, why is the field a `char *` rather than a `const char *`?

Reply:
how do I deallocate the field later on if it's const?
https://github.com/ros2/rclcpp/pull/395#discussion_r154537053

Reply:
How do you know whether or not to deallocate it later (cannot deallocate a literal string...)?